koku/api/provider/serializers.py

```diff
@@ -45,28 +44,64 @@ class ProviderAuthenticationSerializer(serializers.ModelSerializer):
     """Serializer for the Provider Authentication model."""
 
     uuid = serializers.UUIDField(read_only=True)
+
+    # XXX: This field is DEPRECATED;
+    # XXX: the credentials field should be used instead.
     provider_resource_name = serializers.CharField(
         required=True, allow_null=True, allow_blank=True)
 
+    credentials = serializers.JSONField()
+
+    def to_internal_value(self, data):
+        """Convert incoming data to a native value.
+
+        This method is being overridden to handle optional/deprecated fields.
+        """
+        if 'credentials' not in data:
+            data['credentials'] = {}
+
+        if 'provider_resource_name' not in data:
+            data['provider_resource_name'] = 'None'
```

Comment:
Instead of storing the string "None" in the database, I think we should use a check constraint that requires either provider_resource_name or credentials to be null. This would require making both fields nullable, but the check constraint would ensure 1 of them is not null. See: https://docs.djangoproject.com/en/2.2/ref/models/constraints/#checkconstraint

Reply:
Sure thing. That sounds like a much better way to achieve the same goal. Thanks for the suggestion. ^_^